platooning_strategic_IHP/src/platoon_manager_ihp.cpp

```diff
@@ -631,4 +631,63 @@ namespace platoon_strategic_ihp
         // ---> 4.4 return IHP calculated desired speed
         return pos_des;
     }
+
+    // UCLA: find the clsest index to the cut-in joining vehicle.
+    // Note: The index pointing to the gap-leading vehicle's index.
+    int PlatoonManager::getClosestIndex(double joinerDtD)
+    {   
+        /*
+            A naive way to find the closest member to the jioning vehicle.
+            Note: The potential gap leading vehicle should be in front of the joiner (i.e., gap leading vehicle's dtd > joiner's dtd).
+                  If the joiner is already in front of leader --> all loop result will be negative --> return -1 (i.e., cut-in front) 
+        */
+
+        double min_diff = 99999.000;
+        int cut_in_index = -1;
+        // Loop through all platoon members  
+        for(int i = 0; i < platoon.size(); i++) 
+        {
+            double current_member_dtd = platoon[i].vehiclePosition; 
+            double curent_dtd_diff = current_member_dtd - joinerDtD;
+            // update min index
+            if (curent_dtd_diff > 0 && curent_dtd_diff < min_diff)
+                {
+                    min_diff = current_member_dtd;
+                    cut_in_index = i;
+                }
+        }
+        
+        return cut_in_index;
+    }
+
+    // UCLA: find the cut-in join target gap size in downtrack distance (m).
+    double PlatoonManager::getCutInGap(int gap_leading_index, double joinerDtD)
+    {
+        /*
+            Locate the target cut-in join gap size based on the index.   
+        */
+       // cut-in from front 
+       if (gap_leading_index == -1)
+       {
+            double gap_rear_dtd = platoon[0].vehiclePosition;
+            double gap_size = joinerDtD - gap_rear_dtd;
+            return gap_size;
+       }
+       // cut-in from behind 
+       else if (gap_leading_index == platoon.size() + 1)
```

Comment:
This needs to be tested against platoon.size() - 1 since that will be the index of the final vehicle in the platoon (it's where the loop in getClosestIndex() stops).